### PR TITLE
Improve test coverage and some bug fixes

### PR DESCRIPTION
**Pull request by greglandrum**
This mainly adds tests to improve test coverage.

This is, of course, a story which could go on for a long time. Some improvement is better than none. :-)

Bug fixes made along the way:
- copy-pasta in `parseAdjustQueryParametersFromJSON()`
- copy-pasta in `adjustQueryProperties()`
- `Bond::getValenceContrib()` would return non-zero results for atoms not involved in the bond
- `StereoType` missing `operator!=`
- `DeprotectData` missing `operator!=`
- overly aggressive caching in `getDistanceMat()` and `get3DDistanceMat()`

This also deprecates the definition of `BalabanJ` which is in `MolDiscriminators.cpp`. If we want to keep this we should move it to the Descriptors directory.

### CODE REVIEW

File: Code/GraphMol/Bond.cpp [hunk 1]
```diff
@@ -183,6 +183,9 @@ double Bond::getBondTypeAsDouble() const {
 }
 
 double Bond::getValenceContrib(const Atom *atom) const {
+  if (atom != getBeginAtom() && atom != getEndAtom()) {
```
Comment (bp-kelley) [inline Code/GraphMol/Bond.cpp:186]: Should this be a warning?  i.e. atom not in bond?

File: Code/GraphMol/Canon.cpp [hunk 2]
```diff
@@ -18,7 +18,7 @@
 
 namespace RDKit {
 namespace Canon {
-namespace {
+namespace details {
```
Comment (bp-kelley) [inline Code/GraphMol/Canon.cpp:21]: is there a reason to not use an anonymous namespace?  these detail functions don't appear to be used outside the file.

File: Code/GraphMol/molopstest.cpp [hunk 3]
```diff
@@ -418,7 +413,13 @@ void test4() {
   TEST_ASSERT(adjMat[1] == 1);
   TEST_ASSERT(adjMat[2] == 1);
   TEST_ASSERT(adjMat[3] == 0);
-  delete m;
+  bool useBO = true;
+  adjMat = MolOps::getAdjacencyMatrix(*m, useBO);
```
Comment (bp-kelley) [inline Code/GraphMol/molopstest.cpp:417]: Might want to check the prop names are being set properly here between useBO and atom Weights.

**Review by bp-kelley — CHANGES_REQUESTED** [review]
Two small questions, yeoman's work here greg, nice!

File: Code/GraphMol/Bond.cpp (hunk 1 above)
Comment (greglandrum) [reply Code/GraphMol/Bond.cpp:186]: not sure... it's unambiguous that the answer is zero.

File: Code/GraphMol/Canon.cpp (hunk 2 above)
Comment (greglandrum) [reply Code/GraphMol/Canon.cpp:21]: if I leave them in the anonymous namespace then they can't be tested directly and it was much easier to test these directly than to construct test molecules which test them indirectly.

File: Code/GraphMol/molopstest.cpp (hunk 3 above)
Comment (greglandrum) [reply Code/GraphMol/molopstest.cpp:417]: `getAdjacencyMatrix()` doesn't have an option with atom weights.

Comment (bp-kelley) [reply Code/GraphMol/molopstest.cpp:417]: Wrong function, I apparently meant:  getDistanceMat

File: Code/GraphMol/Canon.cpp (hunk 2 above)
Comment (bp-kelley) [reply Code/GraphMol/Canon.cpp:21]: good enough for me.

File: Code/GraphMol/Bond.cpp (hunk 1 above)
Comment (bp-kelley) [reply Code/GraphMol/Bond.cpp:186]: Now that I believe...  I think we can table it for now, I might throw an assertion in for kicks just to see if this every happens internally.

File: Code/GraphMol/molopstest.cpp (hunk 3 above)
Comment (greglandrum) [reply Code/GraphMol/molopstest.cpp:417]: I believe that test does that already

**Pull request merged**